### PR TITLE
2936 The Amount field should be disabled when the investor clicks No

### PR DESCRIPTION
**Pull request by harryng22**

### CODE REVIEW

File: src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx [hunk 1]
```diff
@@ -131,14 +140,16 @@ export const RegisterToInvestStage: React.FC<Props> = (props) => {
...
                   label="How much will be your estimated investment?"
                   trailing={<CurrencyLabel>{props.offer.investingTokenSymbol}</CurrencyLabel>}
+                  value={amount}
                   onChange={(value) => onChangeAmount(value, setFieldValue)}
+                  disabled={!values.isInterested}
                 />
 
                 {errors.amount && <ErrorText>{errors.amount}</ErrorText>}
                 {warning && <ErrorText>{warning}</ErrorText>}
               </FormFieldContainer>
```
Comment (witchaotic) [inline src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx:151]: you need to disable the field, not the submit button. you can still submit with "no"

File: src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx [hunk 2]
```diff
@@ -60,6 +60,7 @@ export const RegisterToInvestStage: React.FC<Props> = (props) => {
   const whitelist = useGetWhitelistStatus(props.offer.id)
   const requestWhitelist = useRequestWhitelist(props.offer.id)
   const disableForm = React.useMemo(() => whitelist.status === WhitelistStatus.accepted, [whitelist])
+  const [amount, setAmount] = React.useState("")
```
Comment (witchaotic) [inline src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx:63]: there is no need for this state, you have values bellow.

File: src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx [hunk 3]
```diff
@@ -80,6 +81,13 @@ export const RegisterToInvestStage: React.FC<Props> = (props) => {
...
 
     setValue('isInterested', value)
+
+    if(!value) {
+      setAmount("")
+      setValue('amount', 0)
+      setWarning("")
+    }
+
   }, [])
```
Comment (witchaotic) [inline src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx:91]: you need to fix the dependencies here

**Review by witchaotic — CHANGES_REQUESTED** [review]

File: src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx (hunk 2 above)
Comment (harryng22) [reply src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx:63]: I used it, but it's not working when I want to set values.amount to 0 in onChangeInterested

File: src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx (hunk 3 above)
Comment (harryng22) [reply src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx:91]: I tried, but it doesn't work

Comment (witchaotic) [reply src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx:91]: why use an empty useCallback. remove the callback then.

**Review by witchaotic — CHANGES_REQUESTED** [review]

File: src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx (hunk 3 above)
Comment (harryng22) [reply src/components/LaunchpadOffer/InvestDialog/content/RegisterToInvest.tsx]: Ok I have removed it

**Review by witchaotic — APPROVED** [review]

**Pull request merged**